### PR TITLE
fix(cli): persist folder trust in sandbox

### LINKED ISSUE #29408: Folder trust choice is not saved when running inside a podman sandbox
### What happened?

When folder trust is enabled and Gemini CLI is running inside a Podman sandbox, choosing Trust folder doesn't seem to persist the choice on the host.
The trust dialog shows up again every time I start Gemini CLI from the same folder.

I reproduced this on a fresh checkout of `main`, without any local changes.
### Setup
1. Enable folder trust in `~/.gemini/settings.json`:
`{ "security": 
   { "folderTrust": 
      { "enabled": true }
    } 
}`
2. Start from a directory that isn't already trusted. It should not be present in `~/.gemini/trustedFolders.json` and should not be under a parent marked as `TRUST_PARENT`.
3. Enable the Podman sandbox from the shell:
`export GEMINI_SANDBOX=podman`

I used `SANDBOX_FLAGS=--userns=keep-id` as well because that's needed for Podman to work correctly on my machine.
### Steps to reproduce
1. Open a normal terminal, `cd` into an untrusted directory, and start Gemini CLI (`gemini`, or `npm start` from a `main` checkout).
2. When Gemini CLI asks **"Do you trust the files in this folder?"**, select **Trust folder**.
3. Quit Gemini CLI.
4. Check `~/.gemini/trustedFolders.json` on the host.
5. Start Gemini CLI again from the same directory.

The host `~/.gemini/trustedFolders.json` is unchanged, so there is no entry for the folder I just trusted.
When I start Gemini CLI again, I get the same trust prompt.

This doesn't happen when I run Gemini CLI without the sandbox.

### What did you expect to happen?

after selecting Trust folder, I expected the folder to be added to the host `~/.gemini/trustedFolders.json`.

on the next launch, gemini cli should recognize the folder as trusted and not show the prompt again.

### Client information

<details>
<summary>Client Information</summary>

Run `gemini` to enter the interactive CLI, then run the `/about` command.

```console
> /about
 CLI Version                   0.60.0                                                    
 Git Commit                    733edcb                                                   
 Model                         Auto                                                      
 Sandbox                       sandbox-0.60.0-d02ae9a30dfd                               
 OS                            linux                                                     
 Auth Method                   gemini-api-key     
```

</details>

### Login information

using API key.

### Anything else we need to know?

Inside the sandbox, `HOME` is set to `/home/node`. The host `~/.gemini` isn't mounted into the container; instead, a temporary directory (`/tmp/gemini-sandbox-*`) is mounted at `/home/node/.gemini`.

when I select Trust folder, the trust entry gets written to that temporary directory. however, `trustedFolders.json` is treated as sensitive and isn't copied back to the host.

the CLI then exits with code `199` to restart. Outside the sandbox, that works because the process is restarted with the saved trust file. with podman, the `199` exit causes the `--rm` container to be removed, so the temporary `.gemini` directory and the trust file are gone. the next launch starts a new container without the trust entry, so the dialog comes back.

I see the same thing when I change the trust through `/permissions`, change the trust for the current folder, and then quit. the host `trustedFolders.json` still doesn't get an entry for the folder.

### PR DESCRIPTION
**Pull request by 21vedansh**
## Summary
When gemini CLI was running in a podman/docker sandbox, the folder trust decision wasn't being saved to the hosT `trustedFolders.json`. Because of that, the trust dialog would show up again every time the CLI was started.

this change handles the initial trust decision on the host before starting the sandbox, and also syncs trust changes made through `/permissions` back to the host.

## Details
`trustedFolders.json` lives on the host and isn't mounted into the sandbox. Previously, selecting **Trust folder** wrote the decision to the temporary `.gemini` directory inside the container. When the CLI exited with code `199`, the `--rm` container was removed along with that file, so the host never got the updated trust entry.

The fix works like this:

1. `ensureHostFolderTrust` checks for folder trust on the host before starting `podman`/`docker` and saves the decision to `~/.gemini/trustedFolders.json`.
2. The trust decision is passed into the sandbox through `GEMINI_CLI_TRUST_WORKSPACE`.
3. Since trust has already been handled on the host, the startup trust dialog isn't shown inside the sandbox.
4. If the trust is changed later through `/permissions`, the sandbox writes the change to `trust-request.json`. When the container exits, the host reads it and updates `trustedFolders.json` if the path belongs to the current workspace.

The change is limited to sandbox + folder trust behavior.

## Related Issues
Fixes #29408

## How to Validate
1. Enable folder trust in `~/.gemini/settings.json`.
2. Use a directory that isn't already in `~/.gemini/trustedFolders.json` and isn't under a `TRUST_PARENT` directory.
3. Enable the sandbox:

   ```bash
   export GEMINI_SANDBOX=podman
I also used `SANDBOX_FLAGS=--userns=keep-id` for my podman setup.
4. Start gemini CLI from the directory (`gemini` or `npm start`).
5. Select Trust folder, quit the CLI, and check `~/.gemini/trustedFolders.json` on the host.
6. Start gemini CLI again from the same directory. The trust dialog should no longer appear.

I also tested changing the trust through `/permissions`, quitting the CLI, and checking that the change was written to the host file.
<!-- List exact steps for reviewers to validate the change. Include commands,
expected results, and edge cases. -->

Unit tests cover:

-  `sandboxTrust`
-  `ensureHostFolderTrust`
-  `useFolderTrust`
-  `usePermissionsModifyTrust`
-  `sandbox.test.ts`
-  `gemini.test.tsx`

## Pre-Merge Checklist

<!-- Check all that apply before requesting review or merging. -->

- [ ] Updated relevant documentation and README (if needed)
- [x] Added/updated tests (if needed)
- [ ] Noted breaking changes (if any)
- [ ] Validated on required platforms/methods:
  - [ ] MacOS
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
    - [ ] Podman
    - [ ] Seatbelt
  - [ ] Windows
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
  - [x] Linux
    - [x] npm run
    - [ ] npx
    - [ ] Docker
    - [x] Podman

### CODE REVIEW

File: packages/cli/src/utils/ensureHostFolderTrust.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  isWorkspaceTrusted,
+  TrustLevel,
+} from '../config/trustedFolders.js';
+import type { LoadedSettings } from '../config/settings.js';
+import { persistHostTrust } from './sandboxTrust.js';
+
+export async function ensureHostFolderTrust(
+  settings: LoadedSettings,
+  cwd: string,
+): Promise<void> {
```
[the remark this answers is not in the document]
Comment (21vedansh) [reply packages/cli/src/utils/ensureHostFolderTrust.tsx:26]: Done. ensureHostFolderTrust now takes isInteractive and returns early when it is false.

File: packages/cli/src/gemini.tsx [hunk 2]
```diff
@@ -612,6 +613,8 @@ export async function main() {
 
       const sandboxArgs = injectStdinIntoArgs(process.argv, stdinData);
 
+      await ensureHostFolderTrust(settings, process.cwd());
```
[the remark this answers is not in the document]
Comment (21vedansh) [reply packages/cli/src/gemini.tsx]: done. the host call now passes partialConfig.isInteractive().

File: packages/cli/src/utils/sandbox.ts [hunk 3]
```diff
@@ -1018,7 +1028,14 @@ export async function start_sandbox(
...
         }
-        resolve(code ?? 1);
+
+        const tmpDir = sandboxTmpDir;
+        const finish = () => resolve(code ?? 1);
+        if (!tmpDir) {
+          finish();
+          return;
+        }
+        void applyTrustRequestFromSandbox(tmpDir, workdir).finally(finish);
```
[the remark this answers is not in the document]
Comment (21vedansh) [reply packages/cli/src/utils/sandbox.ts]: done. Added .catch() so persist failures are logged instead of becoming unhandled rejections. .finally(finish) still runs.
